### ISSUE TITLE
Paste in BOINC does not work in Mac OS

### ISSUE THREAD

**Issue by fckwan** [BOINC/boinc#3409]
I try to add a account manager in BOINC running on Mac OS and the paste (Command-V) does not work for pasting password.

Please make the paste function for account manager password to work.

**Comment by CharlieFenton** [comment]
This has been the case for very many years. I've never been successful figuring out the reason. Perhaps someone else can fix it.

**Comment by ethanteng** [comment]
This was driving me crazy the other day, too! I'll pick it up...

**Comment by ethanteng** [comment]
It appears this is either by design within **wxWidgets**, or it's a bug.

In the code, the password input fields are **wxTextCtrl** classes, created with the style **wxTE_PASSWORD**. For example, see: 

[https://github.com/BOINC/boinc/blob/29f99937f215edc9d48b7d7fc308823d8f05ed36/clientgui/AccountInfoPage.cpp#L190](https://github.com/BOINC/boinc/blob/29f99937f215edc9d48b7d7fc308823d8f05ed36/clientgui/AccountInfoPage.cpp#L190)

In the [wxWidgets documentation](https://docs.wxwidgets.org/3.0/classwx_text_ctrl.html#a5889d0d240e603e32c11bd580b65b3d7), this is intended to make the password input echoed with asterisks, which it does.

However, the side effect of this is that pasting into the field does not work. [Others have reported this](https://forums.wxwidgets.org/viewtopic.php?t=37883), and I've also confirmed it in code, i.e. I changed the style from wxTE_PASSWORD to something else and was then able to paste into the field.

Given that, I'm curious how others feel about this... I _could_ add the functionality to "show / hide" the entered text in the password field; the intent being that when "show" is selected, the user could then paste into the field. However, I'm not sure if the problem is worth the effort?

**Comment by fckwan** [comment]
Hi, 
I am the user reporting this bug. 
I use a password manager to generate very complex password. So it is very difficult for me to type the password instead of cut and  paste.
Regards, 
Boinc user.


 
 
  On Tue, Dec 31, 2019 at 13:02, Ethan Teng<notifications@github.com> wrote:   
It appears this is either by design within wxWidgets, or it's a bug.

In the code, the password input fields are wxTextCtrl classes, created with the style wxTE_PASSWORD. For example, see:

https://github.com/BOINC/boinc/blob/29f99937f215edc9d48b7d7fc308823d8f05ed36/clientgui/AccountInfoPage.cpp#L190

In the wxWidgets documentation, this is intended to make the password input echoed with asterisks, which it does.

However, the side effect of this is that pasting into the field does not work. Others have reported this, and I've also confirmed it in code, i.e. I changed the style from wxTE_PASSWORD to something else and was then able to paste into the field.

Given that, I'm curious how others feel about this... I could add the functionality to "show / hide" the entered text in the password field; the intent being that when "show" is selected, the user could then paste into the field. However, I'm not sure if the problem is worth the effort?

**Comment by samalloing** [comment]
Hi,

Using the mouse (right-click or the similar function on the trackpad) to cut/copy and paste works for me in MacOS. It is just a workaround and not a solution of course.

Sam

**Comment by CharlieFenton** [comment]
I just tried this in the Select Computer dialog with BOINC built from GIT Master, and I was able to paste in a password using `command-v`. I also pasted in a password by doing a control-click on the password field (equivalent to right-click) and selecting `paste`"from the contextual popup menu. So you may want to examine the difference between how that dialog works versus the account manager one. 

One possibility is that something else in that dialog is absorbing the `command-v` keystroke. Please take a look at lines 176-187 in _DlgSelectComputer.cpp_. I added those lines a long time ago to fix the same problem in that dialog.

**Comment by samalloing** [comment]
Yes, that's possible because the command - c is absorbed by Cancel button
